concrete/src/Tree/Node/Type/FileFolder.php

```diff
@@ -31,6 +47,104 @@ public function getTreeNodeTypeName()
         return t('Folder');
     }
 
+    public function getTreeNodeDisplayName($format = 'html')
+    {
+        if ($this->getTreeNodeName()) {
+            $name = tc($this->getTreeNodeTranslationContext(), $this->getTreeNodeName());
+            switch ($format) {
+                case 'html':
+                    return h($name);
+                case 'text':
+                default:
+                    return $name;
+            }
+        } elseif ($this->treeNodeParentID == 0) {
+            return t('Folders');
+        }
+    }
+
+    /**
+     * Returns the storage location id of the folder.
+     *
+     * @return int|null
+     */
+    public function getTreeNodeStorageLocationID()
+    {
+        return (int) $this->fslID;
+    }
+
+    /**
+     * @return \Concrete\Core\Entity\File\StorageLocation\StorageLocation|null
+     */
+    public function getTreeNodeStorageLocationObject()
+    {
+        $app = Application::getFacadeApplication();
+
+        return $app->make(StorageLocationFactory::class)->fetchByID((int) $this->fslID);
+    }
+
+    /**
+     * @param \Concrete\Core\Entity\File\StorageLocation\StorageLocation $storageLocation Storage location object
```

Comment:
According to the conditional, this method also supports other types (https://github.com/concrete5/concrete5/pull/7344/files#diff-65eeada9ffd75ac1f9142813fa390018R93)